### ISSUE TITLE
Curious of the complexity

### ISSUE THREAD

**Issue by jakiestfu** [omidh28/clarifyjs#1]
Thanks for contributing to the open source community something you've found value in!

I'm curious as to why you wouldn't just leverage Classes (or any other flavor, constructor funcs, prototype, etc) to solve the chaining problem?

https://jsfiddle.net/zxf5nho8/

This keeps client code simpler and less abstracted. Thoughts?

**Comment by omidh28** [comment]
True, That's mush more cleaner and simpler. However it doesn't return the right value, Write you call chain not in JavaScript but in mathematics:

```
5 + 3 * 4 - 2 / 3 
```
Put it into your calculator and it returns 16.33 however your program returns 10 instead. Just because mathematics has priorities and does not calculate from left-to-right. So I thought there might be times that we need to reorder our method calls just like math

**Comment by nimatrueway** [comment]
Omid, I gave you a star and we appreciate what you're doing. However, you should keep the global trend of Javascript in mind as well, which is approaching type-safety ("class" concept that @jakiestfu mentioned, or typescript that angularjs has been migrated to, or flow which react has been using etc.)
All in all, I believe we should avoid sacrificing type-safety unless there is a noticeable gain involved.

**Comment by b25** [comment]
in the standard code, you don't want to use this, but on testing or adding control commands to your application, this is a great addition,  thank you :)
